pyqtgraph/functions.py

```diff
@@ -1752,51 +1733,120 @@ def arrayToQPath(x, y, connect='all'):
             if first < len(x):
                 # Replace all non-finite entries from beginning of arr with the first finite one
                 idx[:first] = first
-                arr[1:-1] = arr[1:-1][idx]
+                arr[:] = arr[:][idx]
 
     # decide which points are connected by lines
     if eq(connect, 'all'):
-        arr[1:-1]['c'] = 1
+        polygon = arrayToQPolygonF(
+            arr['x'],
+            arr['y']
+        )
+        path.addPolygon(polygon)
+        return path
     elif eq(connect, 'pairs'):
-        arr[1:-1]['c'][::2] = 0
-        arr[1:-1]['c'][1::2] = 1  # connect every 2nd point to every 1st one
+        if qt6:
+            points = np.insert(arr[['x', 'y']], np.arange(2, len(x), step=2, dtype=int), np.nan, axis=0)
```

Comment:
I find the following to be measurably faster than ```np.insert```:
```python
nans = np.broadcast_to(np.nan, (n // 2, 1))
xs = np.column_stack([arr['x'].reshape((-1, 2)), nans]).ravel()
ys = np.column_stack([arr['y'].reshape((-1, 2)), nans]).ravel()
polygon = arrayToQPolygonF(xs, ys)
```

Reply:
That did knock 10% off of the runtime 🎉

Reply:
Unfortunately this seems to only work with even numbered elements; probably should use `np.pad` to append some extra np.nan values?

Reply:
Could just slice off the last odd-number element from x and y. Surely "pairs" mode requires pairs of values?

Reply:
🤦🏻  why yes, yes it would require "pairs" of values ...